File: EVR/src/main/java/com/group7/evr/service/StationService.java
package com.group7.evr.service;

import com.group7.evr.entity.Station;

import java.util.List;
import java.util.Map;

public interface StationService {
    List<Station> getAllStations();

    List<Station> getNearbyStations(Double lat, Double lng, Double radiusDeg);

    Station createStation(Station station);

    Station updateStation(Integer stationId, Station stationUpdates);

    Map<String, Object> deleteStation(Integer stationId);
    
    void recalculateStationSlots(Integer stationId);
}

File: EVR-FE/src/services/station.service.js
import apiClient from './api/axios';
import { ENDPOINTS } from './api/endpoints';

export const stationService = {
  getStations: async () => {
    return apiClient.get(ENDPOINTS.GET_STATIONS);
  },

  getNearbyStations: async (lat, lng, radiusDeg = 0.02) => {
    return apiClient.get(ENDPOINTS.GET_NEARBY_STATIONS, {
      params: { lat, lng, radiusDeg },
    });
  },

  createStation: async (stationData) => {
    return apiClient.post(ENDPOINTS.CREATE_STATION, stationData);
  },

  updateStation: async (id, stationData) => {
    return apiClient.put(ENDPOINTS.UPDATE_STATION(id), stationData);
  },

  deleteStation: async (id) => {
    return apiClient.delete(ENDPOINTS.DELETE_STATION(id));
  },
};

